### PR TITLE
Validate plugin registry in the settings.json

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
* Refresh the `plugins` view when a new plugin registry is added in the `settings.json`.
* If the new plugin registry is not valid, show a notification about it.
video-convert

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/CRW-1449
fixes https://github.com/eclipse/che/issues/17263
### How to test this PR?

1. Open a workspace from devfile:
```
apiVersion: 1.0.0
metadata:
  name: testnjz3q
components:
  - type: cheEditor
    reference: 'https://raw.githubusercontent.com/chepullreq4/pr-check-files/master/che-theia/pr-1002/che_theia_meta.yaml'
  - id: github/vscode-pull-request-github/latest
    type: chePlugin
```
2. Open `Plugins` view.
3. Open `settings.json` file in the `.theia` directory in the user folder.
4. add a new value to the `chePlugins.repositories` field e.g.
```
{
    "chePlugins.repositories": { "test": "https://test.com" }
}
```
5. The `Plugins` panel must update the plugins list and the notification about the invalid plugin registry must appear.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

**Review by azatsarynnyy — COMMENTED** [review]
I don't see a notification:
Peek 2021-02-23 20-03

**Review by azatsarynnyy — COMMENTED** [review]
I suppose the problem is related to the incorrect image in the proposed devfile - `maxura/che-theia:969`

**Comment by vinokurig** [comment]
Yes, the build wasn't finished when I published the PR, but now it is updated.

**Review by azatsarynnyy — COMMENTED** [review]
I see that it also fixes https://github.com/eclipse/che/issues/17263 :+1:

File: extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts [hunk 1]
```diff
@@ -69,6 +73,53 @@ export class ChePluginManager {
...
+          this.registryList.push({ name: repoName, uri: newPrefs[repoName] });
+        }
+      }
+      // notify that plugin registry list has been changed
+      fireChanged();
+    });
+    this.chePluginServiceClient.onInvalidRegistryFound(async registry => {
+      const result = await this.messageService.warn(
+        `Invalid plugin registry with name "${registry.name}" and url: "${registry.uri}" is detected`,
+        'Open settings.json'
```
[the remark this answers is not in the document]
Comment (vinokurig) [reply extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts:111]: I think it makes sense, fixed.

**Review by azatsarynnyy — APPROVED** [review]

**Pull request merged**